modules/client/src/erc20/internal/client/methods.ts

```diff
@@ -72,45 +76,56 @@ export class ClientErc20Methods extends ClientCore
       throw new Error("A web3 provider is needed");
     }
 
-    // TODO: Remove below as the new contracts are ready
-    await delay(1000);
-    yield {
-      key: ProposalCreationSteps.CREATING,
-      txHash:
-        "0x0123456789012345678901234567890123456789012345678901234567890123",
-    };
-    await delay(3000);
-    yield {
-      key: ProposalCreationSteps.DONE,
-      proposalId:
-        "0x1234567890123456789012345678901234567890123456789012345678901234",
-    };
-
-    // TODO: Uncomment as the new contracts are ready
-
-    /*
-    const erc20VotingInstance = ERC20Voting__factory.connect(
-      this._pluginAddress,
-      signer
+    const erc20Contract = ERC20Voting__factory.connect(
+      _params.pluginAddress,
+      signer,
     );
 
-    const tx = await erc20VotingInstance.newVote(
-      ...unwrapProposalParams(params)
+    let cid = "";
+    try {
+      cid = await this.ipfs.add(JSON.stringify(_params.metadata));
+    } catch {
+      throw new Error("Could not pin the metadata on IPFS");
+    }
+
+    const tx = await erc20Contract.createVote(
+      toUtf8Bytes(cid),
+      _params.actions || [],
+      Math.round(_params.startDate?.getTime() || 0 / 1000),
+      Math.round(_params.endDate?.getTime() || 0 / 1000),
+      _params.executeOnPass || false,
+      _params.creatorVote || 0,
     );
 
-    yield { key: ProposalCreationSteps.CREATING, txHash: tx.hash };
+    yield {
+      key: ProposalCreationSteps.CREATING,
+      txHash: tx.hash,
+    };
 
     const receipt = await tx.wait();
-    const startVoteEvent = receipt.events?.find(e => e.event === "StartVote");
-    if (!startVoteEvent || startVoteEvent.args?.voteId) {
-      return Promise.reject(new Error("Could not read the proposal ID"));
+    const addresslistContractInterface = ERC20Voting__factory.createInterface();
+    const log = receipt.logs.find(
+      (log) =>
+        log.topics[0] ===
+          id(
+            addresslistContractInterface.getEvent("VoteCreated").format(
+              "sighash",
+            ),
+          ),
+    );
+    if (!log) {
+      throw new Error("Failed to create proposal");
+    }
+
+    const parsedLog = addresslistContractInterface.parseLog(log);
+    if (!parsedLog.args["voteId"]) {
+      throw new Error("Failed to create proposal");
     }
 
     yield {
       key: ProposalCreationSteps.DONE,
-      proposalId: startVoteEvent.args?.voteId,
+      proposalId: hexZeroPad(parsedLog.args["voteId"].toHexString(), 32),
```

Comment:
Do we have the certainty that `.args["voteId"]` will always be non-null? If so, that's fine

Reply:
we have a check on line 117